src/core/lender/distributor/switchable.sol

```diff
@@ -42,16 +42,20 @@ contract SwitchableDistributor is Distributor {
     function depend(bytes32 what, address addr) public auth {
         if (what == "currency") {
             currency = CurrencyLike(currency);
-        }  else revert();
+        }  else super.depend(what, addr);
     }
 
     function balance() public {
         if(borrowFromTranches) {
-            uint currencyAmount = add(senior.balance(), junior.balance());
-            borrowTranches(currencyAmount);
+            uint currencyAmount = junior.balance();
+            if (address(senior) != address(0)) {
+                currencyAmount = add(currencyAmount, senior.balance());
+            }
+            _borrowTranches(currencyAmount);
             return;
         }
         uint repayAmount = currency.balanceOf(address(shelf));
-        repayTranches(repayAmount);
+        _balanceTranches();
```

Comment:
would switch the order of these two functions and just call _balanceTranches() if the senior debt() is still not 0 after repayment

Reply:
@charleenfei  no because we don't want the act based on requests from the shelf in that scenario.

Reply:
okay, didn't see the if condition above, cool